python_dwd/additionals/helpers.py

```diff
@@ -111,109 +116,101 @@ def metaindex_for_1minute_data(parameter: Parameter,
                                   parameter.value,
                                   FTP_METADATA_NAME)
 
-    metadata_path = str(metadata_path)
-
     with FTP(DWD_SERVER) as ftp:
         ftp.login()
 
-        metadata_server = ftp.nlst(metadata_path)
+        metadata_filepaths = ftp.list_files(remote_path=str(metadata_path), also_subfolders=False)
 
-    metadata_local = [str(Path(folder,
-                               SUB_FOLDER_METADATA,
-                               metadata_file.split("/")[-1]))
-                      for metadata_file in metadata_server]
+    metadata_filepaths = [create_remote_file_name(file.lstrip(DWD_PATH)) for file in metadata_filepaths]
 
-    metadata_df = pd.DataFrame(None,
-                               columns=METADATA_1MIN_COLUMNS)
+    metaindex_df = pd.DataFrame(None, columns=METADATA_1MIN_COLUMNS)
 
-    for metafile_server, metafile_local in tqdm(zip(metadata_server, metadata_local), total=len(metadata_server)):
-        with FTP(DWD_SERVER) as ftp:
-            ftp.login()
+    metadata_files = Pool().map(download_metadata_file_for_1minute_data, metadata_filepaths)
 
-            ftp.download(metafile_server,
-                         metafile_local)
+    metadata_dfs = Pool().map(combine_geo_and_par_file_to_metadata_df, metadata_files)
 
-        with ZipFile(metafile_local) as zip_file:
-            zip_file_files = zip_file.infolist()
+    metaindex_df = metaindex_df.append(other=metadata_dfs, ignore_index=True)
 
-            zip_file_files = [zip_file_file.filename
-                              for zip_file_file in zip_file_files]
+    columns = metaindex_df.columns
+    META_INDEX_DTYPES = {columns[0]: int,
```

Comment:
You have to take care that the direction of the column is never changing.

Reply:
I replaced the mapping with a predefined dictionary that can be found in constants/column_name_mapping. There I used the names that are defined for the various types of data. Check out the newly pushed verison!

Reply:
Great !